typescript/infra/config/environments/mainnet3/warp/configGetters/getRenzoEZETHWarpConfig.ts

```diff
@@ -82,30 +84,107 @@ const chainProtocolFee: Record<ChainName, string> = {
   zircuit: '400000000000000',
 };
 
-export function getRenzoHook(
-  defaultHook: Address,
-  chain: ChainName,
-  owner: Address,
-): HookConfig {
+export function getRenzoHook(params: {
+  defaultHook: Address;
+  origin: ChainName;
+  destinationChains: ChainName[];
+  owner: Address;
+  useLegacyRoutingHook: boolean;
+}): HookConfig {
+  const {
+    defaultHook,
+    origin,
+    destinationChains,
+    owner,
+    useLegacyRoutingHook,
+  } = params;
+
+  let routingHook: HookConfig;
+
+  if (useLegacyRoutingHook) {
+    // Currently, most of the Hook configs are using the default hook address.
+    // This is here to allow for incremental rollout of the default hook config for other (PZETH, REZ) configs that depend on this Getter.
+    // TODO: Remove this when we replace all the default hook address with HookType.MAILBOX_DEFAULT
+    routingHook = defaultHook;
+  } else {
+    // If origin is blast, use default hook (allows for outbound to all existing chains).
+    // If origin is other chains, use default hook to route to all existing destinations, except blast.
+    routingHook =
+      origin === 'blast'
+        ? {
+            type: HookType.MAILBOX_DEFAULT,
+          }
+        : {
+            type: HookType.ROUTING,
+            owner: owner,
+            domains: destinationChains
+              .filter((c) => c !== origin)
+              .filter((c) => c !== 'blast')
```

Comment:
maybe pull 'blast' out to a constant of `egressOnly` or `outboundOnly` or somethnig